websockify-irc.js

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env node
+
+// A WebSocket to TCP socket proxy
+// Copyright 2012 Joel Martin
+// Licensed under LGPL version 3 (see docs/LICENSE.LGPL-3)
+
+// Known to work with node 0.8.9
+// Requires node modules: ws and optimist
+//     npm install ws optimist
+
+// WEBIRC Spec based on
+// https://ircv3.net/specs/extensions/webirc.html
+
+const argv = require("optimist").argv,
+  net = require("net"),
+  dns = require("dns"),
+  http = require("http"),
+  https = require("https"),
+  url = require("url"),
+  path = require("path"),
+  fs = require("fs"),
+  mime = require("mime-types"),
+  Buffer = require("buffer").Buffer,
+  WebSocketServer = require("ws").Server;
+
+const pjson = require("./package.json");
+
+let webServer,
+  wsServer,
+  source_host,
+  source_port,
+  target_host,
+  target_port,
+  web_path = null,
+  websocket_count = 0;
+
+const options = {
+  family: 6,
+  hints: dns.ADDRCONFIG | dns.V4MAPPED,
+};
+
+// Handle new WebSocket client
+new_client = function (client, req) {
+  websocket_count++;
+  var clientAddr = client._socket.remoteAddress,
+      hostAddr = null;
+
+  console.log(req ? req.url : client.upgradeReq.url);
+
+  console.log(`WebSocket connection from: ${clientAddr}`);
```

Comment:
There's a mixture of writing interpolation like this, and using + to interpolate strings (code consistency), including where code has been added.